File: Practice/Linked List/linkedlist.java
class LinkedList{
    Node head; // head of list

    // LinkedList node. The inner class is made static, so that main() can access it.
    static class Node{
        int data; // Data 
        Node next; // Next

        // Constructor
        Node(int d){
            data = d;
        }
    }

    public void printList(){
        Node n = head;
        while(n!=null){
            System.out.print(n.data + " ");
            n = n.next;
        }
        System.out.println();
    }

    // main() metjod to create simple linked list with 3 nodes
    public static void main(String[] args) {
        // Create an empty linked list
        LinkedList llist = new LinkedList();
        
        llist.head = new Node(1);
        Node second = new Node(2);
        Node third = new Node(3);

        llist.head.next = second; // Link first node to the second node
        second.next = third; // second node to thrid
        third.next = null; // and initialize third node as null

        // Calling the print function
        llist.printList();
    }
}

File: Practice/Linked List/linkedlist.py
# Node class
class Node:
    # Function to initialize the node object
    def __init__(self,data):
        self.data = data
        self.next = None

# This class contains a Node object
class LinkedList:
    # Function to initialize head
    def __init__(self):
        self.head = None
    
    # Print function
    def printList(self):
        temp = self.head
        while(temp):
            print(temp.data,end="-"),
            temp = temp.next
        print("x")
        
# Code execution starts here
if __name__ == "__main__":
    # Start with an empty list
    llist = LinkedList()

    # Assigning values
    llist.head = Node(1)
    second = Node(2)
    third = Node(3)

    # Linking the nodes together
    llist.head.next = second
    second.next = third
    third.next = None

    # Calling the print function
    llist.printList()



